### ISSUE TITLE
Assuring password contains punctuation

### ISSUE THREAD

**Issue by tuxpowered** [soslan/passgen#2]
When supplying limit_punctuation to password generation, is is not guaranteed that the punctuation will be included in the resulting password. This could be classified as a feature or a bug not sure which a user may consider. 

For now, if you want to assure a password contains the special characters from a limited list, you can use this small function. I will likely make an updated patch that does just this shortly.

```
def random_password(length, type='strong'):
    limit_punctuation="!@#$"
    while True:
        password = passgen(length=length, punctuation=True, limit_punctuation=limit_punctuation)
        print(password)
        if any(char in limit_punctuation for char in password):
            break
    return password
```

**Comment by soslan** [comment]
Valid point. This is not a bug because `punctuation=True` just adds punctuation characters to the pool. I wanted to introduce this feature long before. The best way to do it is with `max_punctuation`, `min_punctuation` and `exact_punctuation` arguments.

Also generating passwords from patterns like `Wwwpddd` (Abc@123) solves this problem.

**Comment by soslan** [comment]
This issue has been resolved in 736a343d3ba1877cecdbeb4f0d856ac4eac29d51. Thanks for reporting.
